lib/src/Uncertainty/Distribution/MarshallOlkinCopula.cxx

```diff
@@ -0,0 +1,247 @@
+  computeRange();
+}
+
+/* Comparison operator */
+Bool MarshallOlkinCopula::operator ==(const MarshallOlkinCopula & other) const
+{
+  if (this == &other) return true;
+  return ((alpha_ == other.alpha_) && (beta_ == other.beta_));
+}
+
+Bool MarshallOlkinCopula::equals(const DistributionImplementation & other) const
+{
+  const MarshallOlkinCopula* p_other = dynamic_cast<const MarshallOlkinCopula*>(&other);
+  return p_other && (*this == *p_other);
+}
+
+/* String converter */
+String MarshallOlkinCopula::__repr__() const
+{
+  OSS oss(true);
+  oss << "class=" << MarshallOlkinCopula::GetClassName()
+      << " name=" << getName()
+      << " dimension=" << getDimension()
+      << " alpha=" << alpha_
+      << " beta=" << beta_;
+  return oss;
+}
+
+String MarshallOlkinCopula::__str__(const String & ) const
+{
+  OSS oss(false);
+  oss << getClassName() << "(alpha = " << alpha_ << ", beta = " << beta_ << ")";
+  return oss;
+}
+
+/* Virtual constructor */
+MarshallOlkinCopula * MarshallOlkinCopula::clone() const
+{
+  return new MarshallOlkinCopula(*this);
+}
+
+/* Get the CDF of the distribution */
+Scalar MarshallOlkinCopula::computeCDF(const Point & point) const
+{
+  const Scalar u = point[0];
+  const Scalar v = point[1];
+  const Scalar ualpha = std::pow(u, alpha_);
+  const Scalar vbeta = std::pow(v, beta_);
+  if (ualpha>vbeta)
+  {
+    const Scalar p = std::pow(u, 1-alpha_) * v;
+    return p;
+  }
+  else
+  {
+    const Scalar p = u * std::pow(v, 1-beta_);
+    return p;
+  }
+}
+
+/* Parameters value accessor */
+Point MarshallOlkinCopula::getParameter() const
+{
+  Point point(2);
+  point[0] = alpha_;
+  point[1] = beta_;
+  return point;
+}
+
+void MarshallOlkinCopula::setParameter(const Point & parameter)
+{
+  if (parameter.getSize() != 2) throw InvalidArgumentException(HERE) << "Error: expected 2 value, got " << parameter.getSize();
+  setAlpha(parameter[0]);
+  setBeta(parameter[1]);
+}
+
+/* Tell if the distribution has independent copula */
+Bool MarshallOlkinCopula::hasIndependentCopula() const
+{
+  return ((alpha_ == 1.0) && (beta_ == 1.0));
+}
+
+/* Alpha accessor */
+void MarshallOlkinCopula::setAlpha(const Scalar alpha)
+{
+  if ((alpha <= 0.0) || (alpha >= 1.0)) throw InvalidArgumentException(HERE) << "Alpha MUST be in (0, 1), here alpha=" << alpha;
+  alpha_ = alpha;
+}
+
+/* Alpha accessor */
+Scalar MarshallOlkinCopula::getAlpha() const
+{
+  return alpha_;
+}
+
+/* Beta accessor */
+void MarshallOlkinCopula::setBeta(const Scalar beta)
+{
+  if ((beta <= 0.0) || (beta >= 1.0)) throw InvalidArgumentException(HERE) << "Beta MUST be in (0, 1), here beta=" << beta;
+  beta_ = beta;
+}
+
+/* Beta accessor */
+Scalar MarshallOlkinCopula::getBeta() const
+{
+  return beta_;
+}
+
+/* Get the Kendall concordance of the distribution */
+CorrelationMatrix MarshallOlkinCopula::getKendallTau() const
+{
+  CorrelationMatrix tauKendall(2);
+  tauKendall(1, 0) = alpha_ * beta_ / (alpha_ + beta_ - alpha_ * beta_);
```

Comment:
If alpha= or beta=0 is allowed, tau=0, and if alpha=beta=1, tau=1

Reply:
Ok : added and tested.